src/patcher/java/net/minecraftforge/gradle/patcher/PatcherPlugin.java

```diff
@@ -151,6 +151,10 @@ public void apply(@Nonnull Project project) {
                 m.artifact();
             });
         });
+        project.getRepositories().maven(e -> { // Needed for parchment mappings
```

Comment:
Should we require the end user to add this repo? Or should we add it optionally when they select the parchment mappings channel?
OR should we expose the parchment stuff as a seperate channel provider plugin?
We discussed this in Discord, but the idea is to be generic and impartial to mapping providers.
I'm a little worried about people claiming favoritism if we ship the provider ourselves.

The rest of the API for defining channel providers seems fine, and extensible enough that other services can add onto it.

Reply:
Having a separate plugin for Parchment support seems reasonable if it were to be under the ParchmentMC org so that it could be updated whenever is necessary by the team themselves. This could be converted into a generic PR just adding the support to register it. I am also fine with only conditionally adding the repo if the parchment channel is in use. The benefit of having this under something Parchment controls is that it can be updated whenever, however most likely means less people overall will use it as they would have to edit their buildscripts and add the repo and plugin. But that's the whole point of not having favoritism. ¯\\\_(ツ)\_/¯

Reply:
Yes, I was thinking that they would be the ones to provide and manage the plugin. 
It's a little extra work on the modder's end. But it's just adding a repo/plugin.
The API for adding channel providers is simple enough tho. So if you can finalize that then I think that would be the best approach.
Speak with the  parchment  people and see what they think. At the end of the day it'll be like 2 classes in their plugin.